src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,279 @@
+    eventImages: false,
+  });
+  const [submitted, setSubmitted] = useState<boolean>(false);
+
+  const handleEventName = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventName = event.currentTarget.value;
+    setDatabaseInput({ ...databaseInput, eventName: eventName });
+    setValidationError({
+      ...validationError,
+      eventName: databaseInput.eventName.length < 1,
+    });
+  };
+  const handleEventCategory = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventCategory = event.currentTarget.value;
+    setDatabaseInput({ ...databaseInput, eventCategory: eventCategory });
+    setValidationError({
+      ...validationError,
+      eventCategory: databaseInput.eventCategory.length < 1,
+    });
+  };
+  const handleEventDate = (event: ChangeEvent<HTMLInputElement>) => {
+    const inputDate = new Date(event.currentTarget.value);
+    const currentDate = new Date();
+    const isValidDate = isBefore(inputDate, currentDate);
+    const timestampDate = Timestamp.fromDate(inputDate);
+    setDatabaseInput({
+      ...databaseInput,
+      eventDate: timestampDate,
+    });
+    setValidationError({
+      ...validationError,
+      eventDate: isValidDate,
+    });
+  };
+  const handleEventCapacity = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventCapacity = event.currentTarget.value;
+    if (
+      !eventCapacity.toLowerCase().search(/[a-z]/g || eventCapacity.length < 1)
+    ) {
+      setDatabaseInput({
+        ...databaseInput,
+        eventCapacity: null,
+      });
+      return;
+    }
+    setDatabaseInput({
+      ...databaseInput,
+      eventCapacity: parseInt(eventCapacity),
+    });
+
+    setValidationError({
+      ...validationError,
+      eventCapacity: databaseInput.eventCapacity === null,
+    });
+  };
+  const handleEventDescription = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    const eventDescription = event.currentTarget.value;
+    setDatabaseInput({
+      ...databaseInput,
+      eventDescription: eventDescription,
+    });
+
+    setValidationError({
+      ...validationError,
+      eventDescription: databaseInput.eventDescription.length < 1,
+    });
+  };
+  const handleAddImage = () => {
+    const image = inputValue;
+    if (!inputValue) {
+      return;
+    }
+    setImages((images) => [...images, image]);
+    setDatabaseInput({ ...databaseInput, eventImages: [...images, image] });
+    setInputValue("");
+
+    setValidationError({
+      ...validationError,
+      eventImages: databaseInput.eventImages.length === 0,
+    });
+  };
+  const handleRemoveImage = () => {
+    const updatedImages = [...images];
+    if (updatedImages.length > 0) {
+      updatedImages.pop();
+    }
+    setImages(updatedImages);
+  };
+  const handleInputValue = (event: ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.currentTarget.value);
+  };
+  const handleRequirements = () => {
+    const errorArray = Object.values(validationError).every((value) => {
```

Comment:
You will need to tweak the logic here. I think you want all of the conditions to be false before you submit?